### PR TITLE
Split notes with different nullifiers

### PR DESCRIPTION
**Pull request by PaulLaux**
We need to have different nullifiers for split notes without changing the note commitment.

### CODE REVIEW

File: src/builder.rs [hunk 1]
```diff
@@ -582,8 +589,11 @@ fn partition_by_asset(
     }
 
     if hm.is_empty() {
-        SpendInfo::dummy(AssetBase::native(), rng);
-        hm.insert(AssetBase::native(), (vec![SpendInfo::dummy(AssetBase::native(), rng), SpendInfo::dummy(AssetBase::native(), rng)], vec![]));
+        let dummy_spend = SpendInfo::dummy(AssetBase::native(), rng);
```
Comment (alexeykoren) [inline src/builder.rs:592]: This is not gonna fly because we need different nullifiers for dummy notes, I think

File: src/builder.rs [hunk 2]
```diff
@@ -482,12 +488,12 @@ impl Builder {
                 .cloned()
                 .unwrap();
 
+            let dummy = SpendInfo::dummy(asset, &mut rng);
```
Comment (alexeykoren) [inline src/builder.rs:491]: same here

Comment (PaulLaux) [reply src/builder.rs:491]: so we have this issue in zsa1

Comment (PaulLaux) [reply src/builder.rs:491]:
```rust
           let dummy_spend = spends.first().map_or_else(
                || SpendInfo::dummy(asset, &mut rng),
                |s| s.create_split_spend(&mut OsRng),
            );
            spends.extend(iter::repeat_with(|| dummy_spend.clone()).take(num_actions - num_spends));
```
dummy_spend is cloned

**Pull request closed without merging**